### PR TITLE
Refactor smartquotes

### PR DESCRIPTION
**Pull request by tk0miya**
Before moving SmartQuotes to post transforms, I refactored it as standard Transform module.
This is simply refactoring.

### CODE REVIEW

**Review by jfbu — APPROVED** [review]
Testing looked ok. I checked also with Docutils 0.13.1 (as we monkey patch then docutils.smartquotes.smartchars). But my review is limited by still narrow bounds of my understanding of Sphinx's inner workings...

File: sphinx/ext/autodoc/importer.py [hunk 1]
```diff
@@ -82,15 +82,9 @@ def __getattr__(self, name):
 
 
 class _MockImporter(object):
-    def __init__(self, names):
+    def __init__(self, modules):
```
Comment (tk0miya) [inline sphinx/ext/autodoc/importer.py]: oh, this contains my test code. this is not related with smart quotes...

**Comment by tk0miya** [comment]
Merged. Thank you for reviewing!
